tools/github/autoresponder.ts

```diff
@@ -1,92 +1,20 @@
-/* eslint-disable no-console,import/no-unresolved,import/extensions */
-// @ts-expect-error deno
-import { Octokit } from "https://esm.sh/octokit?dts";
-// @ts-expect-error deno
-const API_SECRET = Deno.env.get("API_SECRET");
-// @ts-expect-error deno
-const GH_SERVICE_ACC_DISCUSSIONS_TOKEN = Deno.env.get("GH_SERVICE_ACC_DISCUSSIONS_TOKEN");
-// @ts-expect-error deno
-const DISCUSSION_NODE_ID = Deno.env.get("DISCUSSION_NODE_ID");
-// @ts-expect-error deno
-const DISCUSSION_BODY = Deno.env.get("DISCUSSION_BODY");
+#!/usr/bin/env node
 
-console.log("discussion body:", DISCUSSION_BODY);
-console.log("discussion node_id:", DISCUSSION_NODE_ID);
+/* eslint-disable no-console */
+import { Octokit } from "@octokit/core";
+import { type AddDiscussionCommentPayload } from "@octokit/graphql-schema";
 
-const octokit = new Octokit({ auth: GH_SERVICE_ACC_DISCUSSIONS_TOKEN });
+import { checkAiAndDiscussionResponses, createCommentBody, getAnswerFromAi, getSimilarDiscussions } from "./utils";
 
-type Discussion = {
-  path: string;
-  content: string;
-  similarity: number;
-  source: string;
-  type: string;
-  meta: Record<string, unknown>;
-  heading: string;
-  slug: string;
-};
+const { OPENAI_API_SECRET } = process.env;
+const { GH_SERVICE_ACC_DISCUSSIONS_TOKEN } = process.env;
+const { DISCUSSION_NODE_ID } = process.env;
+const { DISCUSSION_BODY } = process.env;
 
-const getSimilarDiscussions = async (secret: string, question: string): Promise<string> => {
-  try {
-    const response = await fetch("https://paste.twilio.design/api/discussions-search", {
-      method: "POST",
-      headers: {
-        "Content-Type": "application/json",
-      },
-      body: JSON.stringify({
-        secret,
-        prompt: question,
-      }),
-    });
-    const responseJson = JSON.parse(await response.text());
-
-    const dateFormatter = new Intl.DateTimeFormat("en-US", {
-      year: "numeric",
-      month: "2-digit",
-      day: "2-digit",
-    });
-
-    return (
-      // Get the top 3 results at most
-      responseJson.data
-        .slice(0, 3)
-        // We only want results that are fairly similar, not guesses
-        .filter((item: Discussion) => item.similarity > 0.78)
-        // Convert to markdown
-        .map(
-          (item: Discussion) =>
-            `- [${item.heading}](${item.path}) (updated: ${dateFormatter.format(
-              new Date(item.meta.updatedA as string),
```

Comment:
After all that, the thing that actually broke, was this _teeny-tiny_ typo with the missing t for `updatedAt`

Reply:
🥲